### PR TITLE
Add `castnow` completions

### PR DESCRIPTION
**Pull request by moverest**
Add `castnow` completions (https://github.com/xat/castnow).

### CODE REVIEW

**Comment by faho** [comment]
Please add a comment at the top describing what this tool is, then this'll be good to merge.

File: share/completions/castnow.fish [hunk 1]
```diff
@@ -0,0 +1,30 @@
+set -l __fish_castnow_keys "space\tToggle\ between\ play\ and\ pause m\tToggle\ mute t\tToggle\ subtitles up\tVolume\ up down\tVolume\ down left\tSeek\ backward right\tSeek\ forward n\tNext\ in\ playlist s\tStop\ playback quit\tQuit"
```
Comment (moverest) [inline share/completions/castnow.fish]: castnow_command_completion

I sure did test it. I can remove the quotes if you prefer it without them.

File: share/completions/castnow.fish [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    __fish_complete_suffix .srt
+    __fish_complete_suffix .vtt
+)"
+complete -c castnow -l subtitles-scale -d "Set subtitles font scale" -x
+complete -c castnow -l subtitles-color -d "Set subtitles font RGBA color" -x
+complete -c castnow -l subtitles-port -d "Specify port to be used for serving subtitles" -x
+complete -c castnow -l myip -d "Set local IP" -x
+complete -c castnow -l quiet -d "No output"
+
+complete -c castnow -l type -d "Explicitly set the mime-type" -a "(__fish_print_xdg_mimetypes | string match -r 'video/.*')" -x
```
Comment (moverest) [inline share/completions/castnow.fish]: When I do (in fish version 2.4.0):
```fish
__fish_print_xdg_mimetypes | string match -r '^video/'
```
or

```fish
__fish_print_xdg_mimetypes | string match -r 'video/'
```

I get:
```
video/
video/
video/
video/
video/
video/
video/
```

File: share/completions/castnow.fish (hunk 1 above)
Comment (moverest) [inline share/completions/castnow.fish]: I can see why this worked...

`mount.fish` uses single quotes in `complete` `-a` argument.
```fish
complete -c mount -x -s o --description 'Mount option' -a '(__fish_append , $__fish_mount_opts)'
```
So `$__fish_mount_opts` doesn't get replaced before being passed to `complete`.

Whereas I've used double quotes in `complete`:
```fish
complete -c castnow -l command -d "Execute key command(s)" -x -a "(__fish_append , $__fish_castnow_keys)"
```
So here, `$__fish_castnow_keys` get replaced with its value before being passed to `complete`.

The advantage here is that I can define `__fish_castnow_keys` as a local variable (and not a global one as in `mount.fish`).

So what would you prefer?

Comment (krader1961) [inline share/completions/castnow.fish]: Bingo! Your analysis is correct. I missed the single versus double-quote difference. Thank you for thinking about this and responding to feedback in a thoughtful manner. I prefer the idiom in your change. I think the other, similar, patterns should match this one for consistency. But that's for another issue/pull-request.

**Comment by krader1961** [comment]
Also, you were correct about my comment regarding the use of `.*` in the `string match`. I have a tendency to read that as a boolean grep where the trailing `.*` doesn't make any sense but does make sense in the context of `string match` in fish.

**Comment by krader1961** [comment]
Squashed merged as commit 85212c57.

**Pull request closed without merging**